# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmModule.cpp
# Commit: 6d6adcf77b5a
# Full Hash: 6d6adcf77b5a99316ef743323ccf362972f020dd
# Author: Yury Delendik <ydelendik@mozilla.com>
# Date: 2021-10-22 10:00:52
# Regressor Bug: 1545131
# File Overlap Count: 1
# Description:
#   Bug 1545131 - compress wasm alt data entries (r=jandem)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27992
# ==============================================================================

diff -r a6180f7f6296 -r 6d6adcf77b5a js/src/wasm/WasmModule.cpp
--- a/js/src/wasm/WasmModule.cpp	Thu Oct 21 21:54:52 2021 +0000
+++ b/js/src/wasm/WasmModule.cpp	Thu Oct 21 22:08:14 2021 +0000
@@ -384,14 +384,14 @@
 
 void Module::serialize(const LinkData& linkData,
                        JS::OptimizedEncodingListener& listener) const {
-  auto bytes = MakeUnique<JS::OptimizedEncodingBytes>();
-  if (!bytes || !bytes->resize(serializedSize(linkData))) {
+  Bytes bytes;
+  if (!bytes.resizeUninitialized(serializedSize(linkData))) {
     return;
   }
 
-  serialize(linkData, bytes->begin(), bytes->length());
+  serialize(linkData, bytes.begin(), bytes.length());
 
-  listener.storeOptimizedEncoding(std::move(bytes));
+  listener.storeOptimizedEncoding(bytes.begin(), bytes.length());
 }
 
 /* virtual */
